### ISSUE TITLE
Disable other shape drawing tools when one is selected

### ISSUE THREAD

**Issue by gmarshall56** [geoman-io/leaflet-geoman#958]
Hello:  The request for the behavior of the map is that the user will be allowed to select one and only one shape drawing tool (Polygon, Line, Circle, etc).  So when a shape to draw is selected, the code needs to disable the other shape drawing tool buttons.  

To do this, for example, if the user selected Polyline, my code does the following:
`                        map.pm.disableDraw('Marker');
                        map.pm.disableDraw('Rectangle');
                        map.pm.disableDraw('Polygon');`
...but this does not disable the other shape drawing buttons.  

Can you please advise how I can make this work?
Thank you

**Comment by Falke-Design** [comment]
You need to call 
```
map.pm.addControls({
    drawMarker: false,
// other shapes  ...
})
```

or better `map.pm.setButtonDisabled('drawMarker', true)`

**Comment by gmarshall56** [comment]
Thank you for the code to enter.  I entered the code you suggested in the 
map.on('pm:drawend')
But with this code when the user clicks on any shape tool, if they then change their mind about the shape they selected and they then click the "Cancel" button that shows in the string of buttons that generate for each shape button ("Finish | Remove Last Vertex | Cancel") then all of the other shape buttons that were removed due to your suggested code are removed and do not come back.  Is there another event that I can bind to if they click on the "Cancel" button on a shape drawing button, or perhaps I am not placing your code in the correct map.pm handle?  

Thank you much for your time.

**Comment by Falke-Design** [comment]
Take a look into this example: https://jsfiddle.net/falkedesign/7wtv183q/

```

var btns =  {
      Marker: 'drawMarker',
      Circle: 'drawCircle',
      Polygon: 'drawPolygon',
      Rectangle: 'drawRectangle',
      Polyline: 'drawPolyline',
      Line: 'drawPolyline',
      CircleMarker: 'drawCircleMarker'
};
console.log(btns)

// listen to when drawing mode gets enabled
map.on('pm:drawstart', function(e) {
	console.log(e)
	for(var shape in btns){
  	var btn = btns[shape];
  	if(shape !== e.shape){
    	map.pm.Toolbar.setButtonDisabled(btn, true)
    }
  }
});

// listen to when drawing mode gets disabled
map.on('pm:drawend', function(e) {
	console.log(e)
	for(var shape in btns){
  	var btn = btns[shape];
  	if(shape !== e.shape){
    	map.pm.Toolbar.setButtonDisabled(btn, false)
    }
  }
});
```

Is this what you are searching for?

**Comment by gmarshall56** [comment]
Again, thank you for the code suggestion.  It works fine.  My question is:  when inside the 'pm:drawend' is there a way to determine if the "Cancel" button was clicked?  See I would like the shape buttons (other than the one I selected) _to remain disabled after I finish drawing_, however, if I click the "Cancel" button ( I changed my mind about the shape I selected) then I want the shape buttons(other than the one I selected) to revert back to enabled, using your suggested code.  So, when inside the 'pm:drawend' code I would need to determine if the "Cancel" button was clicked.  If it was clicked then I know I need to revert the shape buttons back to enabled.  If the "Cancel" button was not clicked, then I know to do nothing, leave the buttons disabled.  I hope I am explaining my need clearly.
